sim/core/environment.go

```diff
@@ -226,7 +226,11 @@ func (env *Environment) setupAttackTables() {
 				character := env.Raid.GetPlayerFromUnit(attacker).GetCharacter()
 				weapons := []*Item{character.GetMHWeapon(), character.GetOHWeapon(), character.GetRangedWeapon()}
 
-				if character.Class == proto.Class_ClassMage || character.Class == proto.Class_ClassPriest || character.Class == proto.Class_ClassWarlock || weapons[0] == nil {
+				if weapons[0] == nil {
+					attacker.AttackTables[idx][proto.CastType_CastTypeMainHand] = NewAttackTable(attacker, defender, weapons[0])
+				}
```

Comment:
I think this if can be removed and in the else part of the next if you can just remove the `if weapon != nil` check and create attack tables even for nil weapons. Looking at the `NewAttackTable` function it will work even with a nil weapon as the only place the weapon is used is for getting the weapon skill and that part has its own nil check for the weapon that returns 0 if its nil